specs/PeriodicBackgoundSyncAndBackgroundSync.md

```diff
@@ -0,0 +1,581 @@
+
+**SyncRegistrationManager API**: This API gives developers the ability 
+to manage the periodic background synchronizations and background sync 
+synchronizations in that service worker. 
+
+**PeriodicSyncRegistered API**: This API gives developers the ability 
+to subscribe to the event when a new Periodic Background synchronization 
+task is registered.
+
+**BackgroundSyncRegistered API**: This API gives developers the ability 
+to subscribe to the event when a new Background synchronization task is 
+registered.
+
+**GetSyncRegistrations API**: This asynchronous call returns the 
+collection of all periodic background synchronization registrations or 
+background synchronization registrations based on the 
+`CoreWebView2ServiceWorkerSynchronizationKind` provided. We add
+`CoreWebView2ServiceWorkerSyncRegistrationInfo` with `Tag` and 
+`MinInterval` properties to represent a synchronization registration.
+
+**DispatchPeriodicSyncEvent API**: This API gives developers the ability 
+to trigger a periodic background synchronization task based on the 
+tag name. Developers can use native OS timer to implement a scheduler 
+to trigger periodic synchronization task according to their own logic.
+
+# API Details
+## C++
+```
+/// Indicates the service worker background synchronization type.
+[v1_enum]
+typedef enum COREWEBVIEW2_SERVICE_WORKER_SYNCHRONIZATION_KIND {
+  /// Indicates that the synchronization is a background synchronization.
+  /// See [Background Synchronization](https://developer.mozilla.org/docs/Web/API/Background_Synchronization_API)
+  /// for more information.
+  COREWEBVIEW2_SERVICE_WORKER_SYNCHRONIZATION_KIND_BACKGROUND_SYNC,
+  /// Indicates that the synchronization is a periodic background synchronization.
+  /// See [Periodic Background Synchronization](https://developer.mozilla.org/docs/Web/API/Web_Periodic_Background_Synchronization_API)
+  /// for more information.
+  COREWEBVIEW2_SERVICE_WORKER_SYNCHRONIZATION_KIND_PERIODIC_SYNC,
+} COREWEBVIEW2_SERVICE_WORKER_SYNCHRONIZATION_KIND;
+
+/// This is the ICoreWebView2ServiceWorkerRegistration interface.
+[uuid(08a80c87-d2b7-5163-bce3-4a28cfed142d), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerRegistration : IUnknown {
+  /// Get the synchronization task registration manager for the service worker.
+  /// See ICoreWebView2ServiceWorkerSyncRegistrationManager.
+  ///
+  /// \snippet ScenarioWebView2ServiceWorkerSyncRegistrationManager.cpp SyncRegistrationManager
+  ///
+  [propget] HRESULT SyncRegistrationManager([out, retval] ICoreWebView2ServiceWorkerSyncRegistrationManager** value);
+}
+
+/// This is the ICoreWebView2ServiceWorkerSyncRegistrationManager interface for the
+/// Service Worker Periodic Background Synchronization and Background Synchronization
+/// registration management APIs.
+interface ICoreWebView2ServiceWorkerSyncRegistrationManager : IUnknown {
+  /// Add an event handler for the `BackgroundSyncRegistered` event.

```

Comment:
It will be good to add to this what the event does. Like 'this event is raised when a background sync is registered with this service worker. This can happen when X calls the Y JavaScript method'

Same for the next event